app/src/androidTest/java/org/mozilla/focus/activity/URLCompletionTest.java

```diff
@@ -67,4 +92,192 @@ public void CompletionTest() throws InterruptedException, UiObjectNotFoundExcept
+
+        // Espresso cannot detect the "Already exists" popup.  Instead, check that it's
+        // still in the same page
+        onView(withId(R.id.domainView))
+                .check(matches(isDisplayed()));
+        onView(withId(R.id.save))
+                .check(matches(isDisplayed()));
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Cleanup
+        removeACSite();
+        Espresso.pressBack();
+        toggleAutocomplete();       // Disable autocomplete
+    }
+
+    // exit to the main view from custom autocomplete dialog
+    private static void exitToTop() {
+        Espresso.pressBack();
+        Espresso.pressBack();
+        Espresso.pressBack();
+    }
+
+    private static void toggleAutocomplete() {
+        onView(withText("Add and manage custom autocomplete URLs."))
+                .perform(click());
+    }
+
+    private static void OpenCustomCompleteDialog() {
+        mDevice.waitForIdle();
+        openSettings();
+
+        onData(withTitleText("URL Autocomplete"))
+                .check(matches(isDisplayed()))
+                .perform(click());
+        mDevice.waitForIdle();
+    }
+
+    // Check autocompletion is turned off
+    private static void checkACOff(String url) throws UiObjectNotFoundException {
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText(url);
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals(url));
+        TestHelper.cleartextField.click();
+    }
+
+    // Check autocompletion is turned on
+    private static void checkACOn(String url) throws UiObjectNotFoundException {
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.setText(url.substring(0,2));
+        TestHelper.hint.waitForExists(waitingTime);
+        assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals(url));
+        TestHelper.cleartextField.click();
+    }
+
+
+    private static void removeACSite() {
+        DataInteraction CustomURLRow = onData(anything())
+                .inAdapterView(allOf(withId(android.R.id.list),
+                        childAtPosition(
+                                withId(android.R.id.list_container),
+                                0)))
+                .atPosition(4);
+        CustomURLRow.perform(click());
+        mDevice.waitForIdle();
+        openActionBarOverflowOrOptionsMenu(InstrumentationRegistry.getContext());
+        mDevice.waitForIdle();   // wait until dialog fully appears
+        onView(withText("Remove"))
+                .perform(click());
+        onView(withId(R.id.checkbox))
+                .perform((click()));
+        onView(withId(R.id.remove))
+                .perform((click()));
+    }
+
+    private static void addAutoComplete(String sitename, boolean... checkSuccess) {
```

Comment:
\[PMD\] INFO: The method 'addAutoComplete\(String, boolean\)' has a NCSS line count of 12\.